src/Umbraco.Cms.Integrations.Search.Algolia/Handlers/ContentUnpublishedHandler.cs

```diff
@@ -0,0 +1,57 @@
+﻿using Microsoft.Extensions.Logging;
```

Comment:
Looks like you could usefully introduce a base class for these handlers, that all three inherit from.  That would allow you to move common logic like iterating and filtering the indexes into one place.